elements/pfe-dropdown/test/pfe-dropdown.spec.ts

```diff
@@ -128,43 +140,129 @@ describe('<pfe-dropdown>', function() {
       expect(checkbox.checked).to.be.true;
       checkbox.remove();
     });
-  });
-
-  it(`should set a11y attributes when the disabled attribute is present on the dropdown`, async function() {
-    const element = await createFixture<PfeDropdown>(html`
-      <pfe-dropdown label="Test disabled dropdown" id="disabledDropdown" disabled>
-        <pfe-dropdown-item item-type="link">
-          <a href="#">Link 1</a>
-        </pfe-dropdown-item>
-      </pfe-dropdown>`);
 
-    expect(element.getAttribute('aria-disabled')).to.equal('true');
-
-    element.removeAttribute('disabled');
-    await element.updateComplete;
-
-    expect(element.getAttribute('aria-disabled')).to.equal('false');
-  });
-
-  it('should create dropdown options through addDropdownOptions API', async function() {
-    const element = await createFixture<PfeDropdown>(html`
-      <pfe-dropdown label="Test dropdown with custom options" id="customDropdown">
-      </pfe-dropdown>`);
+    it('should create dropdown options through the options property', async function() {
+      element.options = options;
+      await element.updateComplete;
+      expect(element.children.length).to.equal(options.length);
+    });
 
-    const options: PfeDropdownOption[] = [
-      { href: 'https://bit.ly/3b9wvWg', text: 'Link 1', type: 'link', disabled: false },
-      { href: 'https://bit.ly/3b9wvWg', text: 'Link 2', type: 'link', disabled: false },
-      { href: 'https://bit.ly/3b9wvWg', text: 'Link 3', type: 'link', disabled: true },
-      { type: 'separator' },
-      { text: 'Action 1', type: 'action', disabled: false },
-      { text: 'Action 2', type: 'action', disabled: true },
-    ];
+    it('should add dropdown options through addDropdownOptions API, this should be additive', async function() {
+      const childrenLength = element.children.length;
+      element.addDropdownOptions(options);
+      await element.updateComplete;
+      expect(element.children.length).to.equal(options.length + childrenLength);
+    });
 
-    // setting JS options using addDropdownOptions API
-    element.addDropdownOptions(options);
+    it('should fire the change event when an item is selected', async function() {
+      setTimeout(() => element.querySelector('button')?.click());
+      const event = await oneEvent(element, 'change') as unknown as DropdownChangeEvent;
+      expect(event.action).to.not.be.empty;
 
-    await element.updateComplete;
+      /** @deprecated */
+      setTimeout(() => element.querySelector('button')?.click());
+      const depEvent = await oneEvent(element, 'pfe-dropdown:change') as unknown as CustomEvent;
+      expect(depEvent.detail.action).to.not.be.empty;
+    });
 
-    expect(element.children.length).to.equal(options.length);
+    /** https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html */
+    describe(`keyboard accessibility`, function() {
```

Comment:
_nice_

Reply:
I based this on your accordion work.  Thank you!